Mlem/App/Views/Root/Tabs/Feeds/Feed Posts/FeedPostView.swift

```diff
@@ -33,19 +36,45 @@ struct FeedPostView<EmbeddedContent: View>: View {
         self.overridePostSize = overridePostSize
         self.favoredLink = favoredLink
         self.embeddedContent = embeddedContent
+        self._obscured = .init(wrappedValue: FiltersTracker.main.postWouldBeFiltered(post))
     }
     
     var body: some View {
-        content
-            .contentShape(.interaction, .rect)
-            .contentShape(.contextMenuPreview, .rect(cornerRadius: Constants.main.standardSpacing))
-            .quickSwipes(post.swipeActions(behavior: postSize.swipeBehavior))
-            .contextMenu { post.allMenuActions(
-                showAllActions: false,
-                navigation: navigation,
-                commentTreeTracker: commentTreeTracker
-            ) }
-            .paletteBorder(cornerRadius: postSize.swipeBehavior.cornerRadius)
+        Group {
+            if obscured {
+                obscuredContent
+                    .onTapGesture {
+                        withAnimation {
+                            obscured = false
+                        }
+                    }
+            } else {
+                content
+                    .contentShape(.contextMenuPreview, .rect(cornerRadius: Constants.main.standardSpacing))
+                    .quickSwipes(post.swipeActions(behavior: postSize.swipeBehavior))
+                    .contextMenu { post.allMenuActions(
+                        showAllActions: false,
+                        navigation: navigation,
+                        commentTreeTracker: commentTreeTracker
+                    ) }
+            }
+        }
+        .contentShape(.interaction, .rect)
+        .paletteBorder(cornerRadius: postSize.swipeBehavior.cornerRadius)
+        .onChange(of: filtersTracker.changeHash) {
+            obscured = filtersTracker.postWouldBeFiltered(post)
+        }
+    }
+    
+    @ViewBuilder
+    var obscuredContent: some View {
+        Text("Hidden by user filters")
```

Comment:
Is this slightly better? At least whilst keyword filters are the only type we offer.

```suggestion
        Text("Hidden by keyword filters")
```

Reply:
Sure, that makes sense